include/qdmi/device.h

```diff
@@ -351,7 +351,130 @@ int QDMI_device_session_query_operation_property(
+ * initialized.
+ * @return @ref QDMI_ERROR_FATAL if an unexpected error occurred.
+ * @remark Calling this function with @p value set to @c NULL is expected to
+ * allow checking if the device supports the specified property without
+ * retrieving the property and without the need to provide a buffer for it.
+ * Additionally, the size of the buffer needed to retrieve the property is
+ * returned in @p size_ret if @p size_ret is not @c NULL.
+ * See the @ref QDMI_device_query_pulse_parameter_property documentation for an
+ * example.
+ * @attention May only be called after the session has been initialized with
+ * @ref QDMI_device_session_init.
+ */
+int QDMI_device_session_query_pulse_parameter_property(
+    QDMI_Device_Session session, QDMI_Pulse_Parameter parameter,
+    QDMI_Pulse_Parameter_Property prop, size_t size, void *value,
+    size_t *size_ret);
+/**
+ * @brief Query a pulse waveform property.
+ * @param[in] session The session used for the query. Must not be @c NULL.
+ * @param[in] waveform The pulse waveform to query. Must not be @c NULL.
+ * @param[in] prop The property to query. Must be one of the values specified
+ * for @ref QDMI_Pulse_Waveform_Property.
+ * @param[in] size The size of the memory pointed to by @p value in bytes. Must
+ * be greater or equal to the size of the return type specified for @p prop,
+ * except when @p value is @c NULL, in which case it is ignored.
+ * @param[out] value A pointer to the memory location where the value of the
+ * property will be stored. If this is @c NULL, it is ignored.
+ * @param[out] size_ret The actual size of the data being queried in bytes. If
+ * this is @c NULL, it is ignored.
+ * @return @ref QDMI_SUCCESS if the device supports the specified property and,
+ * when @p value is not @c NULL, the property was successfully retrieved.
+ * @return @ref QDMI_ERROR_NOTSUPPORTED if the device does not support the
+ * property.
+ * @return @ref QDMI_ERROR_INVALIDARGUMENT if
+ *  - @p session or @p waveform is @c NULL,
+ *  - @p prop is invalid, or
+ *  - @p value is not @c NULL and @p size is less than the size of the data
+ *    being queried.
+ * @return @ref QDMI_ERROR_BADSTATE if the property cannot be queried in the
+ * current state of the session, for example, because the session is not
+ * initialized.
+ * @return @ref QDMI_ERROR_FATAL if an unexpected error occurred.
+ *
+ * @remark Calling this function with @p value set to @c NULL is expected to
+ * allow checking if the device supports the specified property without
+ * retrieving the property and without the need to provide a buffer for it.
+ * Additionally, the size of the buffer needed to retrieve the property is
+ * returned in @p size_ret if @p size_ret is not @c NULL.
+ *
+ * See the QDMI_device_query_pulse_waveform_property documentation for an
+ * example.
+ *
+ * May only be called after the session has been initialized with
+ * @ref QDMI_device_session_init.
```

Comment:
```suggestion
 * @remark Calling this function with @p value set to @c NULL is expected to
 * allow checking if the device supports the specified property without
 * retrieving the property and without the need to provide a buffer for it.
 * Additionally, the size of the buffer needed to retrieve the property is
 * returned in @p size_ret if @p size_ret is not @c NULL.
 * See the QDMI_device_query_pulse_waveform_property documentation for an
 * example.
 * @attention May only be called after the session has been initialized with
 * @ref QDMI_device_session_init.
```
to make it in line with the docstring above.